### PR TITLE
feat: multi-prev

### PR DESCRIPTION
**Pull request by AaronGoldman**
multi-prev

### CODE REVIEW

File: CIPs/cip-145.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+cip: 145
+title: Multiple previous (multi-prev)
+author: Aaron Goldman (@aarongoldman), Mohsin Zaidi (@smrz2001)
+discussions-to: <URL to a topic under the CIP category on the Ceramic forum: https://forum.ceramic.network/c/cips>
+status: Draft
+category: <Core | Networking | Interface | RFC | Meta>
+created: 2023-08-21
+edited: 2023-08-21
+requires: <EIP number(s)>
+replaces: <EIP number(s)>
```
Comment (oed) [inline CIPs/cip-145.md:11]:
```suggestion
```

File: CIPs/cip-145.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- `RFC`: an CIP that proposes an implementation standard (i.e. doctypes, document configurations, or document schemas).
+- `Meta`: an CIP that affects the governance process for CIPs.-->
+
+<!--[requires]: A list of CIP(s) that this CIP depends on. *Optional.-->
+
+<!--[replaces]: A list of CIP(s) that this CIP replaces. *Optional.-->
+
+## Simple Summary
+<!--Provide a simplified and layman-accessible explanation of the CIP.-->
+Multi-prev allows the `priv` field in a ceramic event header to be a list of string rather then a string. This is an ordered list and the abrogator may treat the same set of priv CIDs differently depending on the order.
```
Comment (oed) [inline CIPs/cip-145.md:40]:
```suggestion
Multi-prev allows the `prev` field in a ceramic event header to be a list of CIDs rather then a single CID. This is an ordered list and the abrogator may treat the same set of prev CIDs differently depending on the order.
```

These are stored as CID type not string.

File: CIPs/cip-145.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+Rationale goes here.
+
+
+## Backwards Compatibility
+<!--All CIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The CIP must explain how the author proposes to deal with these incompatibilities. CIP submissions without a sufficient backwards compatibility section may be rejected outright.-->
+All events that have a string in `prev` instead of a list will work with this CIP.
+Nodes that don't have this CIP will not understand what new events with a `prev` list means.
```
Comment (oed) [inline CIPs/cip-145.md:75]: So it's a soft fork? New nodes will be able to understand old nodes, but not the other way around.

File: CIPs/cip-145.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+
+## Implementation
+<!--The implementations must be completed before any CIP is given status "Final", but it need not be completed before the CIP is accepted.-->
+Implementation goes here.
+
+legacy `priv`
```
Comment (oed) [inline CIPs/cip-145.md:82]:
```suggestion
legacy `prev`
```

File: CIPs/cip-145.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+```json
+{ "prev" = [CID, CID, CID]}
+```
+
+multi-priv `priv`
```
Comment (oed) [inline CIPs/cip-145.md:87]:
```suggestion
multi-priv `prev`
```

File: CIPs/cip-145.md (hunk 3 above)
Comment (smrz2001) [reply CIPs/cip-145.md]: Yes, new nodes should work with everything that already exists.

Please note that new data even on existing streams might be aggregated differently on newer nodes compared to older nodes.

File: CIPs/cip-145.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
+---
+cip: 145
+title: Multiple previous (multi-prev)
+author: Aaron Goldman (@aarongoldman), Mohsin Zaidi (@smrz2001)
+discussions-to: https://forum.ceramic.network/t/cip-145-multiple-previous-multi-prev/1277
+status: Draft
+category: <Core>
```
Comment (oed) [inline CIPs/cip-145.md:7]:
```suggestion
category: Core
```

**Review by oed — APPROVED** [review]
Formatting looks good to me. I approve merging this draft

**Pull request merged**